frontend/src/services/notebookService.ts

```diff
@@ -290,24 +286,31 @@ export const startNotebook = (data: StartNotebookData): Promise<Notebook> => {
   });
 };
 
-export const stopNotebook = (projectName: string, notebookName: string): Promise<Notebook> => {
+export const stopNotebook = async (
+  projectName: string,
+  notebookName: string,
+): Promise<Notebook> => {
   const dateStr = new Date().toISOString().replace(/\.\d{3}Z/i, 'Z');
-  const patch: RecursivePartial<Notebook> = {
+  const notebook = await getNotebook(projectName, notebookName);
+  const patch: Notebook = _.merge({}, notebook, {
     metadata: { annotations: { 'kubeflow-resource-stopped': dateStr } },
-  };
-
-  return patchNotebook(projectName, notebookName, patch);
+  });
+  return replaceNotebook(projectName, notebookName, patch);
 };
 
-export const patchNotebook = (
+export const replaceNotebook = async (
   projectName: string,
   notebookName: string,
-  updateData: RecursivePartial<Notebook>,
+  data: Notebook,
 ): Promise<Notebook> => {
   const url = `/api/notebooks/${projectName}/${notebookName}`;
+  const notebook = await getNotebook(projectName, notebookName);
+  notebook.spec = data.spec;
+  notebook.metadata.annotations['kubeflow-resource-stopped'] =
```

Comment:
Could not find a cleaner way to do this, the 'kubeflow-resource-stopped' must be undefined in case of a start, `' '` - does not work.